### DISCUSSION TITLE
Shutterbutton not working

### DISCUSSION THREAD

**Discussion by crllsn** [arendst/Tasmota#16198] [Q&A] [answered]
Hi, perhaps it's just me but I really cannot figure it out.
I configured a Sonoff Dual R3 as a shutter controller. Everything is fine but I cannot make shutterbutton work (I have button in wall, not switches)
In the default template for Dual R3 I have GPIO32 and GPIO33 assigned to Switch 1 and Switch 2.
I think the problem is that: shutterbutton works with buttons not switches.
Someone discussed about thisin the past, but no solution has been found...
Please note that if I change switch with button in template editor and I press the button, nothing happens.
Honestly I read everything I could, but I did not find any solution. Probably there's something I do not understand...
Any idea?

Thanks!

**Comment by crllsn** [comment, marked as answer]
Ok I figured it out.
Shutterbutton function NEEDS button configured IN TEMPLATE and NOT switches. (I'll change wiki since it's not so clear even if it can look obvious).

@stefanbode Do you think it could be possible to achieve this behaviour (supposing to have push button on the wall):
1) during "tilt" period the button act as a button (e.g. I press, it tilts; I release, it stops).
2) after the tilt period the button act as defined in shutterbutton

Probably a "new" shutterbutton function could be added to have this? something like a 'p' (pulse function)?

For example:

`Shutterbutton1 2 p 100 50 0`

Single press: it moves as long as it is pressed
2x press: moves to 100%
3x moves to 50%
no mqtt

Obviously if this could help others and not just me!

**Reply by stefanbode** [reply]
Can you define a bit more what is your definition of "in tilt mode" and "outside tilt mode". Try to think as a developer that needs to create a " if then else" statement.

**Reply by stefanbode** [reply]
There is a "long press" event on buttons. Maybe we can use this one for the first idea. But be aware there is some initial delay to decide if it is just a long short press or really a long press. Please see in the documentation of button events

**Reply by crllsn** [reply]
The shutterbutton 'p' modification can be good (even if there's no distinction of tilting/not tilting - moving, it's confortable to use single press to tilt and then move as long as you press, when you release it stops). Then if you want "automatic" movement, you 2x or 3x press the button.
Thinking and discussing, now I think this could be the best solution in terms of ergonomy.
I do not know if it's easy or not in terms of developing.
What do you think?

**Reply by stefanbode** [reply]
You will never know until you try. I will check what I got back for the buttons on long press and if I got a message on relieving the button .

**Reply by crllsn** [reply]
I'm just afraid that my first proposal is too difficult to use. I think that modifying shutterbutton, having "single press"="simple push button" is more convenient if you have to tilt and possibly manually move a venetian. It's almost impossible pushing to start the tilt and then pushing again to stop the tilt. 2 seconds is too short time to have a satisfactory result. It's a pain to use...
Unless I'm not able to configure buttons in the correct way...

**Reply by stefanbode** [reply]
So, two hours later I'm a bit more educated about buttons. The hold event is there and there is a way to use this to switch on the relay and let the tilt move. But I do NOT get an info back from TASMOTA, when the user released the button. Therefore I have no chance to react on this an stop. Additionally there are some side effects on "reset 1" if you hold a button more than 40 seconds, but this can be switched off. The missing release  message is a problem. Maybe there is another way to get this info....

**Reply by crllsn** [reply]
Again: I do not know what's behind, so maybe it's a stupid idea.
If we could use shutterbutton function with switches (I mean SWITCH in template - now you can't), my proposed 'p' function could be a simple "switchmode 0" used with tasmota SWITCH and wall buttons. Till now I was using my venetian like this, so I can be fairly precise in tilt and position, but it's boring I cannot use multiple press to have 50% and 100% open...

**Reply by stefanbode** [reply]
Ok, there is a pull request open: https://github.com/arendst/Tasmota/pull/16229
With this change the HOLD event will stop with the release of the button. No further change required. Just define the HOLD e.g. 0 or 100 and then if you hold the button the shutter will move until you release the button. All other events like single, double and so on presses still works.

**Reply by stefanbode** [reply]
My tested ENV:
ShutterButton1 1 100 50 74 100
ShutterButton1 2 0 50 24 0

**Reply by crllsn** [reply]
Great!! Now everything works as expected (by me at least!). Simple and smooth solution!
FYI I set 

`SetOption32 5`

to minimize delay in tilting slats.

**So shortly if you have venetian blinds with push buttons on wall the solution is:
using tasmota >12.0.2.4 define BUTTONS in template, define shutterbuttons with HOLD option and you can benefit of manually moving the blind (e.g. to be precise with tilt) holding the button or you can use 1x, 2x, 3x short presses to move the blind to set positions.**

I'll be back from holidays in a couple of week an I'll add some configuration advice to tasmota wiki to make everything clearer to newbies like me!
Thanks again!!
